api-guidelines/v4.0/API-DESIGN-GUIDELINES/DISCOVERY-API.md

```diff
@@ -0,0 +1,170 @@
+2.0](../../../api-specifications/discovery/2.0-draft/).
+
+## Root-Level Information
+
+The root-level information (in this example, `version`, `informationalVersion`,
+`suite`, and `build`) describes the deployed application. If the application
+complies with these design guidelines and the relevant API specification, then
+most _client applications_ should not need to know which _server_ application it
+is communicating with. However, this information can be valuable when debugging
+the client-server interaction.
+
+## Data Models
+
+This section lists all of the Data Models that are served by this API
+application. This will always include the core Ed-Fi (Unifying) Data Model.
+Other models listed are the _extensions_ that a given deployment implements. The
+example above shows the core Teacher Preparation Data Model (TPDM) extension.
+
+## URLs Section
+
+The URLs section helps client applications discover for themselves how to
+interact with the Ed-Fi API application. The _required_ URLs are further
+described in the following sections.
+
+The optional values listed below represent features of the the Ed-Fi ODS/API
+Platform. Other applications can add their own URLs as needed.
+
+```json
+"xsdMetadata": "https://api.ed-fi.org/v7.0/api/metadata/xsd",
+"changeQueries": "https://api.ed-fi.org/v7.0/api/changeQueries/v1/",
+"composites": "https://api.ed-fi.org/v7.0/api/composites/v1/",
+"identity": "https://api.ed-fi.org/v7.0/api/identity/v2/"
+```
+
+### Dependencies
+
+All implementations of an Ed-Fi API _should_ expose an endpoint for dependency
+metadata.  This endpoint will provide developers with the data needed to load
+resources in the appropriate order. The endpoint _should_ have a JSON
+implementation and optional a GraphML implementation.  A sample of the JSON
+output is below:
+
+```json
+[
+  {
+    "resource": "/ed-fi/absenceEventCategoryDescriptors",
+    "order": 1,
+    "operations": [
+      "Create"
+    ]
+  },
+  {
+    "resource": "/ed-fi/academicHonorCategoryDescriptors",
+    "order": 1,
+    "operations": [
+      "Create"
+    ]
+  },
+  …
+  {
+    "resource": "/tpdm/surveySectionResponsePersonTargetAssociations",
+    "order": 22,
+    "operations": [
+      "Create"
+    ]
+  }
+]
+```
+
+### OpenApiMetadata
+
+An Ed-Fi API implementation _should_ declare itself in OpenAPI; the version of
+OpenAPI is up to the implementation. The OpenAPI specification must be a
+faithful representation of all available resources supported by the API. A given
+API application might support multiple specifications, for example one for the
+core Ed-Fi data model resources, and one for Ed-Fi descriptors. The
+specifications provided by an API application _must_ be consistent with the
+official [Ed-Fi API Specifications](../../../api-specifications/).
+
+### OAuth
+
+The OAuth URL must be the token URL used by clients for authentication. The URL
+is not required to be on the same base address, i.e. it could be hosted on
+another server.
```

Comment:
Is this true of all URLs or only OAuth (guessing all)? Also missing _must_ italics and should probably _could_ for the base address

Reply:
Only true of this one - with the notion that someone could be using a third-party OAuth provider. I'll update accordingly.